Cargo.toml

```diff
@@ -119,11 +119,7 @@ harness = false
 members = [
     "pyo3-macros",
     "pyo3-macros-backend",
-    "examples/pyo3-benchmarks",
-    "examples/pyo3-pytests",
-    "examples/maturin-starter",
-    "examples/setuptools-rust-starter",
-    "examples/word-count"
+    "examples"
```

Comment:
Is this change for simplicity?

One downside of this is that CI will be slower because it'll build a new fresh set of dependencies in a unique target dir for each example. We might be able to hack around that by forcing the target directory via `CARGO_TARGET_DIR`.

Another downside is that whole-workspace commands like `cargo fmt` and `cargo clippy --all-targets --workspace` won't catch the example and pytest crates any more. Opened #2047 which should resolve that (if this PR then updates the `Makefile` to ensure everything is covered).

Reply:
That's a good question, and I'm not sure the current approach is best.

My motivation for this is that I'd rather split all this example related stuff off from pyo3/cargo.toml. We could inline it in the pyo3/cargo.toml, but this means we have to put a lot of stuff there:

About 50-100 lines of this, depending on how many examples we end up with:
```
[[examples]]
name = ...
path = ...
```

To make example discovery work, we need to add dev dependencies to support any dependencies we only need for examples, so we end up with something like this:

```
[dev-dependencies]
assert_approx_eq = "1.1.0"
criterion = "0.3.5"
trybuild = "1.0.49"
rustversion = "1.0"
# 1.0.0 requires Rust 1.50
proptest = { version = "0.10.1", default-features = false, features = ["std"] }
serde_json = "1.0.61"

# Only needed for examples
rayon = "1"
bytes = "1"
..etc
```

I would rather be explicit about this and put this in an example crate.

For comparison, it could end up looking like [tokio](https://github.com/tokio-rs/tokio/blob/master/examples/Cargo.toml)'s example folder.

Reply:
Ah ok, understood, and I think this is very nice 👍 

I've added a commit which tweaks CI to re-use the same target dir.